src/tests_deprecated/unit/engines/gna/i16_quantisation_test.cpp

```diff
@@ -13,20 +13,43 @@
 #include "ngraph_functions/builders.hpp"
 
 using namespace InferenceEngine;
-using namespace GNAPluginNS;
+using namespace ov::intel_gna::frontend;
 using namespace GNATestIRs;
 
 class I16QuantisationTest : public GNATest<> {
  protected:
-    ModelQuantizer q;
-    LayersQuantizer lc = LayersQuantizer(1.0f);
-
-    InferenceEngine::CNNLayerPtr  quantize (InferenceEngine::CNNLayerPtr lp) {
+    InferenceEngine::CNNLayerPtr quantize (InferenceEngine::CNNLayerPtr lp) {
         auto newLayer = InferenceEngine::injectData<QuantizedLayerParams>(lp);
-        transformLayer(newLayer, lc);
+        GNAPluginNS::Config gna_config;        
+        gna_config.gnaPrecision = InferenceEngine::Precision::I16;
+        gna_config.gnaFlags.input_low_precision = false;
+        LayerQuantizer lq(gna_config);
+        lq.quantize(*newLayer);
         return newLayer;
     };
 
+    InferenceEngine::CNNNetwork quantize(const InferenceEngine::CNNNetwork& model, float scale_factor) const {
+        auto scale_factors = std::vector<float>({scale_factor});
+
+        GNAPluginNS::GnaInputs inputs;
+        InferenceEngine::InputsDataMap inputs_map = model.getInputsInfo();
+
+        if (inputs_map.size() > 1) {
+            THROW_GNA_EXCEPTION << "Scale factors are not set for some of the inputs";
```

Comment:
In this particular case I'd leave it as it is, because the exception is meant to signal the user the model he created is not supported by the test.

Reply:
It is test method not a library method, and exception type suggest issue in library not in test setup.

Is it on done on purpose that GNA specific exception is used in case model cannot by handled by tests?

Reply:
As agreed this check will be removed and replaced by a function name change.